lib/ControlSystemsBase/src/types/conversion.jl

```diff
@@ -298,18 +299,40 @@ function siso_ss_to_zpk(sys, i, j)
     return z, eigvals(A), k
 end
 
+function charpoly(A::AbstractMatrix{T}) where {T}
+    N = size(A, 1)
+    poly_factors::Vector{T} = vec(ones(T, N+1, 1))
```

Comment:
```suggestion
    poly_factors = vec(ones(T, N+1))
```
the type assertion is only beneficial if type inference fails